addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -0,0 +1,290 @@
+
+import base64
+import json
+import logging
+import requests
+from werkzeug.exceptions import Forbidden
+
+from odoo import http, tools, _
+from odoo.addons.iap.tools import iap_tools
+from odoo.http import request
+
+_logger = logging.getLogger(__name__)
+
+
+class MailPluginController(http.Controller):
+
+    @http.route('/mail_client_extension/modules/get', type="json", auth="outlook", csrf=False, cors="*")
+    def modules_get(self, **kwargs):
+        """
+            deprecated as of saas-14.3, not needed for newer versions of the mail plugin but necessary
+            for supporting older versions
+        """
+        return {'modules': ['contacts', 'crm']}
+
+    @http.route('/mail_plugin/partner/enrich_and_create_company',
+                type="json", auth="outlook", cors="*")
+    def res_partner_enrich_and_create_company(self, partner_id):
+        """
+        Route used when the user clicks on the create and enrich partner button
+        it will try to find a company using IAP, if a company is found
+        the enriched company will then be created in the database
+        """
+        response = {}
+
+        partner = request.env['res.partner'].browse(partner_id)
+
+        normalized_email = partner.email_normalized
+        if not normalized_email:
+            response = {'error': 'contact has no email'}
+            return response
+
+        company_domain = normalized_email.split('@')[1]
+
+        company, enrichment_info = self._create_company_from_iap(company_domain)
+
+        response['enrichment_info'] = enrichment_info
+        response['company'] = self._prepare_company_values(company)
+        if company:
+            partner.write({'parent_id': company})
+
+        return response
+
+    @http.route(['/mail_client_extension/partner/get', '/mail_plugin/partner/get']
+        , type="json", auth="outlook", cors="*")
+    def res_partner_get(self, email=None, name=None, partner_id=None, **kwargs):
+        """
+        returns a partner given it's id or an email and a name.
+        In case the partner does not exist, we return partner having an id -1, we also look if an existing company
+        matching the contact exists in the database, if none is found a new company is enriched and created automatically
+
+        old route name "/mail_client_extension/partner/get is deprecated as of saas-14.3,it is not needed for newer
+        versions of the mail plugin but necessary for supporting older versions
+        """
+        response = {}
+
+        if not (partner_id or (name and email)):
+            return {'error': _('You need to specify at least the partner_id or the name and the email')}
+
+        if partner_id:
+            partner = request.env['res.partner'].browse(partner_id)
+            return self._prepare_contact_values(partner)
+
+        normalized_email = tools.email_normalize(email)
+        if not normalized_email:
+            response['error'] = 'Bad email.'
```

Comment:
Strange, either return a key, either a translated string but seems a mix of both ?

Reply:
This is also based on the old code, after looking at the code of the legacy version of the plugin, it seems that we can use a translated string here, since the plugin only verifies if there is an "error" key in the returned dictionary.

Reply:
Ok cool :)